app/layersmodel.h

```diff
@@ -17,81 +11,33 @@
 #ifndef LAYERSMODEL_H
 #define LAYERSMODEL_H
 
-#include <QAbstractListModel>
-#include <QList>
-#include <QSet>
+#include <QObject>
 
-class QgsQuickMapSettings;
-class QgsMapLayer;
-class QgsProject;
+#include "qgsmaplayermodel.h"
+#include "qgsmaplayer.h"
+#include "qgsvectorlayer.h"
```

Comment:
Whenever possible, I prefer to keep #includes outside of header files and only use forward declarations... for example do we need to #include "qgsvectorlayer.h" here?